bluesky_widgets/models/search.py

```diff
@@ -270,9 +271,11 @@ def ensure_abs(*abs_or_rel_times):
 
 
 class SearchInput:
-    def __init__(self, *, text_search_supported=False):
+    def __init__(self, *, fields=None, text_search_supported=False):
         self._since = None
         self._until = None
+        self._fields = fields or []
+        self._field_search = UpdateOnlyDict({field: None for field in self._fields})
```

Comment:
Therre a handy classmethod on `dict` for constructing things like this

```suggestion
        self._field_search = UpdateOnlyDict({dict.fromkeys(self._fields))
```